### ISSUE TITLE
Add Likert/matrix question format

### ISSUE THREAD

**Issue by ei8fdb** [nextcloud/forms#356]
Which Forms version are you running?
v2.0.0 beta 1 (build 20200505)

Nextcloud or ownCloud?:
Nextcloud
## Feature request
Add Likert/matrix question format

## Rational
Having a way for a user to choose a scale answer is helpful when trying to gather data on a scale. 

A Likert scale as [described by wikipedia](https://en.wikipedia.org/wiki/Likert_scale) is:
> a psychometric scale commonly involved in research that employs questionnaires. It is the most widely used approach to scaling responses in survey research.

## UI
There's lots of ways of doing this. Here's one option:

![Screenshot from 2020-05-06 17-00-38](https://user-images.githubusercontent.com/1899506/81199965-40f9f200-8fbb-11ea-981f-713e0d448573.png)

There are many options on how to display this. I've not thought too uch yet about it, but can help with designs if it helps.

## Priority
High

As a user researcher who creates questionnaires having questions that can gather ranking data is very important - however I can understand this might be complicated/a lot of work to implement. There's also a lot of ways to implement this so I think it requires time to discuss.

Thoughts?

**Comment by skjnldsv** [comment]
@jancborchardt

**Comment by ei8fdb** [comment]
(Also: apologies - I didn't see there was a feature request template already there. Hence my malformed feature requests!)

**Comment by skjnldsv** [comment]
> (Also: apologies - I didn't see there was a feature request template already there. Hence my malformed feature requests!)

I literally just added it because I could see all the incoming requests ;)
You did good, no worries :hugs:

**Comment by jancborchardt** [comment]
Yep, we should absolutely have this, Google Forms has both:
- Multiple choice grid
- Checkbox grid

**Comment by ei8fdb** [comment]
@jancborchardt Let me know if I can help here too.

**Comment by jotoeri** [comment]
just for reference - Google Screenshot for this can be found here: https://github.com/nextcloud/forms/issues/555#issuecomment-711094687

**Comment by TheBitFighter** [comment]
As asked I would like to propose a variant of this question type where instead of text, the radio buttons form a linear scale from a set start number to a set end number. The ends of this scale have a description text probably beneath them. What I'm describing is basically the question type "Linear scale" from Google forms. We use this a lot in our post event evaluation forms. This would either be a variation of the originally proposed matrix or a derived question type.
![linear scale](https://user-images.githubusercontent.com/22227120/113473715-8e9d0a00-946b-11eb-8318-9510880ee74c.png)
Sorry for the German.
Cheers!

**Comment by kojid0** [comment]
Leaving this screenshot here: It could also be displayed as stars.
![grafik](https://user-images.githubusercontent.com/24477557/149328786-cad88a17-2c20-4054-9e47-5d3a95037693.png)

**Comment by stefan-franz** [comment]
That question type would very helpful for me.

**Comment by sirdrakeistda** [comment]
It would be cool to not only use radio buttons, but also implement sliders for scale input. The results could be visualised as histogram.

**Comment by Flo-112** [comment]
Hi @all,
is there anybody working on this issue.

Thats one of my greatest NC Problems....
Greats
Flo

**Comment by jancborchardt** [comment]
@Chartman123 yeah, I would say it makes a lot of sense, as it's very useful for more complex surveys and research.

I know @nimishavijay would really appreciate it. :)

**Comment by nimishavijay** [comment]
Definitely agreed and glad to see this feature get picked up! I'm happy to help out with design wherever needed :)

**Comment by Flo-112** [comment]
hey all, 
would be great if the would cooming soon. i am looking forward to the day.

Did anybody know i which percentage is the progress?

Edit: i just saw that there is another threat:  [#2945](https://github.com/nextcloud/forms/pull/2945)

**Comment by Koc** [comment]
I guess we can close this issue as https://github.com/nextcloud/forms/pull/2945 got merged, can't we?

**Comment by Chartman123** [comment]
No, rating is still missing. Basically linear scale, but with stars/hearts/whatever instead of radio buttons
